src/api/alchemy-websocket-provider.ts

```diff
@@ -1008,13 +1003,57 @@ function addToPastEventsBuffer<T>(
   pastEvents.push(event);
 }
 
-function isAlchemyEvent(event: AlchemyEventType): event is object {
+function isAlchemyEvent(
+  event: AlchemyEventType
+): event is AlchemyPendingTransactionsEventFilter {
   return typeof event === 'object' && 'method' in event;
 }
 
-function getAlchemyEventTag(event: AlchemyEventType): string {
+/**
+ * Creates a string representation of an `alchemy_pendingTransaction`
+ * subscription filter that is compatible with the ethers implementation of event tag.
+ *
+ * @example
+ *   ```js
+ *   // Returns 'alchemy:0xABC:0xDEF|0xGHI:true'
+ *   const eventTag =  getAlchemyEventTag(
+ *   {
+ *     "method": "alchemy_pendingTransaction",
+ *     "fromAddress": "0xABC",
+ *     "toAddress": ["0xDEF", "0xGHI"],
+ *     "hashesOnly: true
+ *   });
+ *   ```;
+ *
+ * @param event
+ * @internal
+ */
+export function getAlchemyEventTag(event: AlchemyEventType): string {
   if (!isAlchemyEvent(event)) {
     throw new Error('Event tag requires AlchemyEventType');
   }
-  return 'alchemy:' + (('address' in event && event.address) || '*');
+  const fromAddress = serializeAddressField(event.fromAddress);
+  const toAddress = serializeAddressField(event.toAddress);
+  const hashesOnly = serializeBooleanField(event.hashesOnly);
+  return 'alchemy:' + fromAddress + ':' + toAddress + ':' + hashesOnly;
+}
+
+function serializeAddressField(
+  field: string | Array<string> | undefined
+): string {
+  if (field === undefined) {
+    return '*';
+  } else if (Array.isArray(field)) {
+    return field.join('|');
+  } else {
+    return field;
+  }
+}
+
+function serializeBooleanField(field: boolean | undefined): string | undefined {
+  if (field === undefined) {
+    return '*';
+  } else {
+    return field ? 'true' : 'false';
```

Comment:
This is fine, but can also be written `return field.toString()` or `return field + ""`.